src/main/java/pl/agh/edu/iisg/io/vmms/vmmsbackend/controller/VMPoolController.java

```diff
@@ -43,46 +43,113 @@ public VMPoolController(VMPoolService vmPoolService) {
         return vmPoolService.findByDescriptionContaining(tag);
     }
 
+
+    @RequestMapping(value = "/update", method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.CREATED)
+    public String updateFromFile(@RequestParam("file") MultipartFile file) throws HttpException {
+        int vmPoolsNumber = 0;
+        if (!file.isEmpty()) {
+            vmPoolsNumber = updateVmPools(file, vmPoolsNumber, false);
+        }
+
+        return "Updated " + vmPoolsNumber + " virtual machine pools.";
+    }
+
     @RequestMapping(value = "/import", method = RequestMethod.POST)
     @ResponseStatus(HttpStatus.CREATED)
     public String importFromFile(@RequestParam("file") MultipartFile file) throws HttpException {
 
         int vmPoolsNumber = 0;
         if (!file.isEmpty()) {
-            try {
-                String line;
-                BufferedReader bufferedReader = new BufferedReader(
-                        new InputStreamReader(file.getInputStream()));
-
-                VMPoolCSVParser parser = new VMPoolCSVParser();
-
-                while ((line = bufferedReader.readLine()) != null) {
-                    Optional<VMPool> o = parser.parseLine(line);
-                    if(o.isPresent()) {
-                        System.out.println("parsed");
-                        try{
-                            vmPoolService.save(o.get());
-                        }catch (Exception e){
-                            String details = "";
-                            if(e.getMessage().contains("constraint [uk_")){
-                                details = " - Unique key";
-                            }else if(e.getMessage().contains("constraint [fk_")){
-                                details = " - Foreign key";
-                            }
-                            throw new VMPoolImportFileException("Constraint not met" + details);
+            vmPoolsNumber = updateVmPools(file, vmPoolsNumber, true);
+        }
+
+        return "Created " + vmPoolsNumber + " virtual machine pools.";
+    }
+
+    private int updateVmPools(@RequestParam("file") MultipartFile file, int vmPoolsNumber, boolean replace) throws VMPoolImportFileException {
```

Comment:
Is the RequestParam annotation necessary here?

Reply:
I don't like http path mapping and update logic mixed up in one class. I'd extract the update logic to external class + maybe link it with some service.